### PR TITLE
Suzanne's edits

### PR DESCRIPTION
**Pull request by sascala**
They are substantial and not necessarily correct. Hence the separate branch.

<<Replace this line with your Revision Title>>

Summary:
For more details follow http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html.

JIRA issues:

### CODE REVIEW

**Comment by janisz** [comment]
@sascala Can you split long lines. It will make reviewing further changes easier.

File: docs/docs/networking.md [hunk 1]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+#### Host Mode Use Case
+
+When in host mode, the container will have the host network interface defined in the container namespace. The container will not do any port mapping and the application will open up port `:8080` on the host network interface. In this case, there is no container port and the `hostPort` is `:8080`.
+
+#### Bridge Mode Use Case
+
+If the container is run in bridge mode, it gets its own network interface and the `containerPort` is `:8080`. In bridge mode, the container will open a pseudo random port on the host and bridge/NAT communication to the container port.
+
+For example, lets say the host port port is `:31000`. Now, the `containerPort` is `:8080`, but the `hostPort` is `:31000`. Clients to this service will open port `:31000` on the host the service is running on. The container will route traffic to the internal port in the container, `:8080`.
```
Comment (janisz) [inline docs/docs/networking.md]: Why do we need `:` before the port number?

File: docs/docs/networking.md [hunk 2]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+In Marathon, this is referred to as the service port. The service port is specified in the `servicePort` parameter of the ACME app definition. Let us assume `servicePort` is specified as `:8080`.
+
+The service port is metadata; Marathon does not do anything with this information except track it and provide it to a load balancer. The user is expected to configure a load balancer to host a port `:8080`, or refer to the `servicePort`, and route to each of the instances of the application. All metadata is queryable from Marathon.
+
+The [Marathon-LB service](https://docs.mesosphere.com/1.10/networking/marathon-lb/), when configured, does exactly this.  Marathon-LB will register all instances of an app and route to its configured `servicePort`. In the configuration in this example, a client will connect to a load balancer at port `:8080` (`servicePort`), which will route (with some algorithm) to `10.0.0.2:31000` (`hostPort`), which will in turn route to `:8080` (`containerPort`) of the internal application.
```
Comment (janisz) [inline docs/docs/networking.md]: `with some algorithm`? can you explain? Is it load balancing algorithm?

File: docs/docs/networking.md [hunk 3]
```diff
@@ -111,8 +137,9 @@ Now let's say you have 3 instances of App ACME, for example 10.0.0.2:31000, 10.0
         }
     }]
 ```
-This configuration will create a FQN of `acme.marathon.l4lb.thisdcos.directory` at port :10000 which will load balance all instances of the application.  It is important to note that Marathon itself doesn't do anything with this configuration it is metadata that Marathon manages for the application.  A client in DC/OS could open a connection to `acme.marathon.l4lb.thisdcos.directory` at port :10000, which would route to 10.0.0.2:31000 which will route to the `containerPort` of :8080.
+This configuration will create a FQN of `acme.marathon.l4lb.thisdcos.directory` at port `:10000`, which will load balance all instances of the application.
```
Comment (janisz) [inline docs/docs/networking.md]: `FQN` →`FQDN` 
please add a link and/or full text

**Comment by sascala** [comment]
@janisz @janisz I tried to split the long paragraphs of the previous version into shorter sections. Is this what you mean? Happy to help once I understand what you'd like.

File: docs/docs/networking.md (hunk 1 above)
Comment (sascala) [reply docs/docs/networking.md]: @kensipe What was the rationale there?

File: docs/docs/networking.md (hunk 2 above)
Comment (sascala) [reply docs/docs/networking.md]: @kensipe Is the idea with the "some algorithm" language that the exact algorithm will be installation-specific and not necessarily something that can be documented more generally?

File: docs/docs/networking.md (hunk 3 above)
Comment (sascala) [reply docs/docs/networking.md]: 👍

**Comment by ichernetsky** [comment]
@sascala I think what @janisz meant is to make lines visually shorter (by inserting new lines to make them no longer than 80 or 120 characters), but this has been discussed previously and we decided to go with long lines because this way it is easier to review changes, due to the way diff highlighting works.

File: docs/docs/networking.md (hunk 2 above)
Comment (ichernetsky) [reply docs/docs/networking.md]: Yes. Basically things like Marathon-LB use it for routing/load-balancing, and it is up to an external routing/load-balancing service what algorithm to use.

File: docs/docs/networking.md [hunk 4]
```diff
@@ -111,8 +137,9 @@ Now let's say you have 3 instances of App ACME, for example 10.0.0.2:31000, 10.0
         }
     }]
 ```
-This configuration will create a FQN of `acme.marathon.l4lb.thisdcos.directory` at port :10000 which will load balance all instances of the application.  It is important to note that Marathon itself doesn't do anything with this configuration it is metadata that Marathon manages for the application.  A client in DC/OS could open a connection to `acme.marathon.l4lb.thisdcos.directory` at port :10000, which would route to 10.0.0.2:31000 which will route to the `containerPort` of :8080.
+This configuration will create a fully qualified domain name (FQDN) of `acme.marathon.l4lb.thisdcos.directory` at port `:10000`, which will load balance all instances of the application.
```
Comment (ichernetsky) [inline docs/docs/networking.md]: The phrase `... name ... at port` is slightly incorrect, in a sense that a domain name has nothing to do with a port. I think it should be re-phrased.

**Comment by ichernetsky** [comment]
Hi @janisz Could you please take another look at it taking into account my comments?

File: docs/docs/networking.md (hunk 2 above)
Comment (sascala) [reply docs/docs/networking.md]: I clarified.

File: docs/docs/networking.md (hunk 1 above)
Comment (sascala) [reply docs/docs/networking.md]: ...removed them

File: docs/docs/networking.md (hunk 4 above)
Comment (sascala) [reply docs/docs/networking.md]: I clarified

File: docs/docs/networking.md [hunk 5]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+The following use cases explore how you can configure port mapping to meet your needs. 
 
-`VIP_0` defined in `portDefinitions` as a label is like `servicePort` in that it is informational and implementation dependent.  However, when used with DC/OS, services internal to DC/OS will make available to the cluster that DNS name and port as a routing to services.  An example configuration:
+You will configure 3 defined ports (`containerPort`, `hostPort`, and `servicePort`) and the label for `VIP_0` in your port mapping.
+
+All applications on Mesos are hosted in containers. Containers can be run in bridge or host mode.
```
Comment (kensipe) [inline docs/docs/networking.md]: Containers can run with a network configuration in bridge or host mode.

File: docs/docs/networking.md [hunk 6]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+Let us assume an app called "ACME," and that the actual application opens port `8080`.
```
Comment (kensipe) [inline docs/docs/networking.md]: Let us assume an application called "ACME" which opens port `8080`.

File: docs/docs/networking.md [hunk 7]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+#### Host Mode Use Case
+
+When in host mode, the container will have the host network interface defined in the container namespace. The container will not do any port mapping and the application will open up port `8080` on the host network interface. In this case, there is no container port and the `hostPort` is `8080`.
+
+#### Bridge Mode Use Case
+
+If the container is run in bridge mode, it gets its own network interface and the `containerPort` is `8080`. In bridge mode, the container will open a pseudo random port on the host and bridge/NAT communication to the container port.
+
+For example, lets say the host port port is `31000`. Now, the `containerPort` is `8080`, but the `hostPort` is `31000`. Clients to this service will open port `31000` on the host the service is running on. The container will route traffic to the internal port in the container, `8080`.
```
Comment (kensipe) [inline docs/docs/networking.md]: "host port port" -> "host port"

Comment (kensipe) [reply docs/docs/networking.md]: Clients to this service ->   Clients connecting to this service

File: docs/docs/networking.md [hunk 8]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+##### Multiple App Instances Use Case
+
+Now, let's say you have 3 instances of app ACME: `10.0.0.2:31000`, `10.0.0.2:31001`, and `10.0.0.3:31000`. There is a common need to have a port that will route with an algorithm to each of those instances. The algorithm used is configured in the load balancer. This "port" is metadata specific to this application.  All instances of the application will be hosted behind this port.
```
Comment (kensipe) [inline docs/docs/networking.md]: The algorithm used is configured in the load balancer. ->
The algorithm used is determined by the configuration of the load balancer.

File: docs/docs/networking.md [hunk 9]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+In Marathon, this is referred to as the service port. The service port is specified in the `servicePort` parameter of the ACME app definition. Let us assume `servicePort` is specified as `8080`.
+
+The service port is metadata; Marathon does not do anything with this information except track it and provide it to a load balancer. The user is expected to configure a load balancer to host a port `8080`, or refer to the `servicePort`, and route to each of the instances of the application. All metadata is queryable from Marathon.
```
Comment (kensipe) [inline docs/docs/networking.md]: The user is expected to configure a load balancer to host a port `8080` ->
The DevOps team setting up this service is expected to create a script or provide a means to read the `servicePort` and configure the load balance to route calls to that port ( port `8080` in this case) to each of the instances of the application.

File: docs/docs/networking.md [hunk 10]
```diff
@@ -94,13 +94,39 @@ See [the port definition section](#port-definition) for more information.
...
+
+The [Marathon-LB service](https://docs.mesosphere.com/1.10/networking/marathon-lb/), when configured, does exactly this.  Marathon-LB will register all instances of an app and route to its configured `servicePort`. In the configuration in this example, a client will connect to a load balancer at port `8080` (`servicePort`), which will route (with an algorithm) to `10.0.0.2:31000` (`hostPort`), which will in turn route to `8080` (`containerPort`) of the internal application.
```
Comment (kensipe) [inline docs/docs/networking.md]: (https://docs.mesosphere.com/1.10/networking/marathon-lb/), when configured, does exactly this.   Marathon-LB is an HAProxy service with scripts that will register all instances of an app and route to its configured `servicePort`.

**Pull request merged**